jarviscli/plugins/dns_lookup.py

```diff
@@ -0,0 +1,51 @@
+from plugin import plugin, alias
+import socket
+from colorama import Fore
+
+
+# a reverse DNS lookup or reverse DNS resolution (rDNS)
+# is the querying technique of the Domain Name System (DNS)
+# to determine the domain name associated with an IP address
+@plugin("dns reverse")
+def hostname_lookup(jarvis, s):
+    while True:
+        # asks for an IP address
+        ip = str(jarvis.input("Please input an IP address: "))
+        try:
+            # tries to find the hostname for given IP address
+            jarvis.say("The hostname for that IP address is: " +
+                       str(socket.gethostbyaddr(ip)[0]), Fore.BLUE)
+            break
+        except Exception as e:
```

Comment:
Just noticed: As a rule, never catch 'Exception'.

For example, let's imagine a typo 'Fore.BUE'. Your function will work without showing a error - but whatever you input - valid or invalid IP - it will always say 'Please make sure you are inputing a valid IP'.

Let's look at socket documentation:

https://docs.python.org/3.5/library/socket.html#exceptions

You should make it a:

``except socket.gaierror as e``

Because that's what is returned if ip not found. Additionally you could catch ``OSError`` in a second except clause. This would include e.g. ``socket.timeout`` - and you could tell the user, something really unexpected happened - but IP address might be right.

Reply:
Makes sense

Reply:
According to the documentation, socket.gaierror would only catch errors from address-related errors by getaddrinfo() and getnameinfo(). and i am using neither. 

socket.herror would catch errors from  gethostbyname_ex() and gethostbyaddr() so I think that is the correct one, what do you think?

Reply:
Ok so after some testing socket.gaierror does catch errors from socket.gethostbyname but neither socket.herror or OSError is catching errors from socket.gethostbyaddr...based on this I will as Exception for now?